raiden/api/python.py

```diff
@@ -607,14 +607,15 @@ def transfer_and_wait(
         """ Do a transfer with `target` with the given `amount` of `token_address`. """
         # pylint: disable=too-many-arguments
 
-        async_result = self.transfer_async(
+        payment_status = self.transfer_async(
             registry_address=registry_address,
             token_address=token_address,
             amount=amount,
             target=target,
             identifier=identifier,
         )
-        return async_result.wait(timeout=transfer_timeout)
+        payment_status.payment_done.wait(timeout=transfer_timeout)
```

Comment:
`wait` will return `None` in case of an exception or timeout, `True` as we set it in the raiden_event_handler.
In:
https://github.com/raiden-network/raiden/blob/d50045c7aec544d225dcddc4fb2ee81f35e1949e/raiden/api/rest.py#L1023-L1029

In case of a timeout/exception, the value for the AsyncResult is going to be None and payment will fail:
https://github.com/raiden-network/raiden/blob/d50045c7aec544d225dcddc4fb2ee81f35e1949e/raiden/api/rest.py#L1041-L1046

So in conclusion, i believe this is a cleaner solution than the one we had in #3362